app/assets/javascripts/components/live_search.js

```diff
@@ -0,0 +1,94 @@
+(function($) {
+  $.fn.live_search = function(results, reset) {
+    new LiveSearch(this, results, reset);
+    return this;
+  }
+
+  LiveSearch = function(el, results, reset) {
+    this.el = el;
+    this.results = results;
+    this.reset = reset;
+    this.form = this.el.closest('form');
+    this.url = this.form.attr('action');
+
+    var self = this;
+
+    this.el
+      .on('focus', function() { self.select() })
+      .on('input', function() { self.search() });
+
+    this.select = function() {
+      this.el.select();
+    };
+
+    this.search = function() {
+      this._cancel_debounced();
+
+      if (!this.el.val().length) {
+        this._reset_results();
+      } else {
+        this._execute_search();
+      }
+    };
+
+    this._reset_results= function() {
+      if (this._has_reset_node()) {
+        this._hide_results();
+      } else {
+        this._execute_search();
+      }
+    };
+
+    this._execute_search = function() {
+      this._debounce(function() {
+        $.get(self.url, self.form.serialize()).always(function(response) {
+          var parsed_response = self._parse_response(response);
```

Comment:
Identifier 'parsed_response' is not in camel case.<br>Identifier '_parse_response' is not in camel case.